.github/workflows/code_coverage.yml

```diff
@@ -0,0 +1,48 @@
+name: Code Coverage
+on:
+  pull_request:
+    branches-ignore: "dependabot/**"
+jobs:
+  comment-coverage-difference:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: go test -v -coverpkg=./... -coverprofile=profile.cov ./...
+      - id: coverage-on-this-pr
+        run: |
+          echo "pr-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - run: go test -v -coverpkg=./... -coverprofile=profile.cov ./...
+      - id: coverage-on-main # TODO: can consider if we should save this coverage as an artifact?
```

Comment:
yeah, not sure if it's worth the extra dev time, I think it's fine to calculate it each time for now

Reply:
I was wondering which one of the following we should do:
* expect the package.json to have a script called `yarn test:coverage`
* directly use `jest --coverage`

Kind of related question: should we use `jest --coverage --changedSing=origin/main`?

Reply:
> expect the package.json to have a script called yarn test:coverage

That's a nice thing to expect, so people can reproduce this check easily.

> Kind of related question: should we use jest --coverage --changedSing=origin/main?

I don't think so. That would only load some test files (and functional files) so the coverage report would not be accurate.

Reply:
oops got it now, thanks!